### ISSUE TITLE
Make `tx_abort` a variant of `InteractiveTxMessageSend` and refactor 

### ISSUE THREAD

**Issue by dunxen** [lightningdevkit/rust-lightning#3406]
Shouldn't hold up this PR, but is there any reason why we don't simply add a `TxAbort` variant to `InteractiveTxMessageSend`? Then we wouldn't need either `InteractiveTxMessageSendResult` or `HandleTxCompleteResult`, IIUC, which would remove two layers of structs.

Otherwise, we go through the following conversions just to end up with a `MessageSendEvent`:
- `Result<InteractiveTxMessageSend, AbortReason>`
- `InteractiveTxMessageSendResult(Result<InteractiveTxMessageSend, msgs::TxAbort>)`
- `MessageSendEvent`

The `Result`s don't seem to buy us anything more than the extra variant would as we aren't doing anything special for the `Err` case. Then it would simply be:
- `InteractiveTxMessageSend`
- `MessageSendEvent`

And AFAICT, `InteractiveTxConstructor` has the `channel_id`, so there's no need to grab it from the `ChannelContext`. Or will it change in some case?

_Originally posted by @jkczyz in https://github.com/lightningdevkit/rust-lightning/pull/3137#discussion_r1838980862_

**Comment by jkczyz** [comment]
FYI, dual funding will slip to 0.3 as development has been focused on splicing.

**Comment by wpaulino** [comment]
This may not be relevant anymore. After #4060, `InteractiveTxMessageSendResult` was removed and `Result<InteractiveTxMessageSend, AbortReason>` is now used to fail the ongoing negotiation internally.

**Comment by wpaulino** [comment]
@jkczyz is there anything else you want to improve here?

**Comment by TheBlueMatt** [comment]
Dual funding is gonna slip yet again, setting to 0.4

**Comment by jkczyz** [comment]
> [@jkczyz](https://github.com/jkczyz) is there anything else you want to improve here?

Yeah, looks like this is no longer relevant as per your earlier comment.
